### DISCUSSION TITLE
存量 Processor 插件迁移到 ProcessorV2 接口

### DISCUSSION THREAD

**Discussion by liuhaoyang** [alibaba/ilogtail#784] [Ideas]
目前 pipeline v2 已经完整支持 `metrics` `trace` `log` 数据模型，但只有`processor_rename`  `processor_split_log_string` `default` 等有限的几个processor插件实现了 v2 接口。

近期我计划对现有的v1 Processor 开始逐步迁移，也欢迎社区的同学一起参与来加速这个工作。

使用这个 discussion 用来追踪存量 Processor 插件支持v2实现的进度。

**Comment by messixukejia** [comment]
把文档在这里也链接下？

**Reply by liuhaoyang** [reply]
> 把文档在这里也链接下？

迁移文档需要写一下，我先完成这个吧

**Comment by pj1987111** [comment]
我可以来改，这里记录一下更改情况，有需要认领的同学可以在最右加上自己的id：

- [ ] processor_add_fields
- [ ] processor_anchor
- [ ] processor_appender
- [ ] processor_base64_decoding
- [ ] processor_base64_encoding
- [ ] processor_cloud_meta
- [ ] processor_csv
- [x] processor_default @snakorse #624
- [ ] processor_desensitize
- [ ] processor_dict_map
- [ ] processor_drop
- [ ] processor_drop_last_key
- [ ] processor_encrypt
- [ ] processor_fields_with_condition
- [ ] processor_filter_key_regex
- [ ] processor_filter_regex
- [ ] processor_geoip
- [ ] processor_gotime
- [ ] processor_grok
- [x] processor_json @yyuuttaaoo #984 
- [ ] processor_md5
- [ ] processor_packjson
- [ ] processor_pick_key
- [ ] processor_regex
- [x] processor_rename @snakorse #624
- [ ] processor_split_char
- [ ] processor_split_key_value
- [ ] processor_split_log_regex
- [x] processor_split_log_string @liuhaoyang #647 
- [ ] processor_split_string
- [ ] processor_string_replace
- [ ] processor_strptime

**Reply by liuhaoyang** [reply]
点赞

**Reply by pj1987111** [reply]
希望能给几个单测的例子，我看目前的结构不大好做单测的模拟

**Reply by yyuuttaaoo** [reply]
可以看一下我的提交 #983
